### PR TITLE
PYTHON-1320 Remove legacy CRUD methods

### PR DESCRIPTION
**Pull request by ShaneHarvey**
Remove save, insert, update, remove, and find_and_modify.

### CODE REVIEW

File: test/test_bulk.py [hunk 1]
```diff
@@ -295,8 +295,8 @@ def test_upsert(self):
         self.assertEqual(self.coll.count_documents({'foo': 'bar'}), 1)
 
     def test_numerous_inserts(self):
-        # Ensure we don't exceed server's 1000-document batch size limit.
-        n_docs = 2100
```
Comment (behackett) [inline test/test_bulk.py:299]: I don't remember why this was hard coded. Do we not get it from the server on some old version like 2.6?

File: test/test_collection.py [hunk 2]
```diff
@@ -1941,13 +1957,79 @@ def test_min_query(self):
         self.assertEqual(2, docs[0]["x"])
 
     def test_numerous_inserts(self):
-        # Ensure we don't exceed server's 1000-document batch size limit.
+        # Ensure we don't exceed server's maxWriteBatchSize size limit.
         self.db.test.drop()
-        n_docs = 2100
```
Comment (behackett) [inline test/test_collection.py:1946]: Same question here.

File: test/test_monitoring.py [hunk 3]
```diff
@@ -1088,49 +865,46 @@ def test_insert_many(self):
 
     def test_legacy_insert_many(self):
```
Comment (behackett) [inline test/test_monitoring.py:866]: Seems like you should remove "legacy" from the name.

File: tools/benchmark.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2009-2015 MongoDB, Inc.
```
[the remark this answers is not in the document]
Comment (behackett) [reply tools/benchmark.py:1]: This benchmark isn't great and never really has been. Let's just get rid of it.

**Review by behackett — CHANGES_REQUESTED** [review]

**Review by prashantmital — APPROVED** [review]

File: tools/benchmark.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply tools/benchmark.py:1]: Removed.

File: test/test_bulk.py (hunk 1 above)
Comment (ShaneHarvey) [reply test/test_bulk.py:299]: We always use max_write_batch_size internally, this test was never updated when the server updated the batch size from 1000 to 100000.

**Review by behackett — APPROVED** [review]

**Pull request merged**